pkg/csv2lp/csv2lp_test.go

```diff
@@ -204,6 +204,40 @@ func Test_CsvToLineProtocol_SkipRowOnError(t *testing.T) {
 	require.Equal(t, messages, 2)
 }
 
+// Test_CsvToLineProtocol_RowSkipped tests that error rows are reported to configured RowSkippedListener
+func Test_CsvToLineProtocol_RowSkipped(t *testing.T) {
+	var buf bytes.Buffer
+	log.SetOutput(&buf)
+	oldFlags := log.Flags()
+	log.SetFlags(0)
+	oldPrefix := log.Prefix()
+	prefix := "::PREFIX::"
+	log.SetPrefix(prefix)
+	defer func() {
+		log.SetOutput(os.Stderr)
+		log.SetFlags(oldFlags)
+		log.SetPrefix(oldPrefix)
+	}()
+
+	csv := "sep=;\n_measurement;a|long:strict\n;1\ncpu;2.1\ncpu;3a\n"
+
+	reader := CsvToLineProtocol(strings.NewReader(csv)).SkipRowOnError(true)
+	reader.RowSkipped = func(src *CsvToLineReader, err error, _row []string) {
+		log.Println(err, string(src.Comma()))
```

Comment:
The other tests in this library intercept the calls to `log` in order to validate the `reader` calls `log` appropriately.
Here `log.Println(...)` is being called explicitly in your test.

Could we change this to gather up all the arguments passed to each call to `RowSkipped` and explicitly check they are as expected. Rather than intercepting `log` calls and ensuring that it is just called three times.

i.e. specifically assert that each of the following contents is within the error:
```
line 3: column '_measurement': no measurement supplied
line 4: column 'a': '2.1' cannot fit into long data type
line 5: column 'a': strconv.ParseInt: parsing "3a": invalid syntax
```

Reply:
The test also checks that no warning messages are logged when RowSkipped is set (using a logger). Yes, explicit validattion of RowSkipped parameters will improve the readability of the test.